src/main/scala/firrtl/transforms/FlattenRegUpdate.scala

```diff
@@ -0,0 +1,117 @@
+// See LICENSE for license details.
+
+package firrtl
+package transforms
+
+import firrtl.ir._
+import firrtl.Mappers._
+import firrtl.Utils._
+
+import scala.collection.mutable
+
+object FlattenRegUpdate {
+
+  /** Mapping from references to the [[Expression]]s that drive them */
+  type Netlist = mutable.HashMap[WrappedExpression, Expression]
```

Comment:
We really need to formalize this into an analysis.

Reply:
Agreed, it pops up all over the place